### PR TITLE
Make "diff copy" feature work with lazy-loaded diffs

### LINKED ISSUE #377: Issue with "Easier copy-pasting from diffs by making +/- signs unselectable"
It doesn't seem to be working all the time with all the pages. Sometimes I see both the "unselectable +" and the "selectable +". I haven't found a pattern to reproduce the issue but, I can link you to a page in which I was facing the same issue. Here's a link to it, and a screenshot as well. If you need anything else, let me know.

[Issue](https://github.com/DmitryMalkovich/stock-hawk-app/commit/13709ac740fe82fb5d607f90c78cd490f5686093)

[Screenshot](https://www.dropbox.com/s/qjhvrxpb7763eff/Screenshot%20%2817%29.png?dl=0)

### PR DESCRIPTION
**Pull request by jgierer12**
Fixes #377. ~Definitely not the nicest solution to re-run the function on scroll, but I can't think of any other solution.~ ~Also not sure how to handle https://github.com/sindresorhus/refined-github/issues/377#issuecomment-296781677~

### CODE REVIEW

**Comment by sindresorhus** [comment]
`$(document).on('scroll')` is a recipe for scroll jank. I really don't want that. Look into using Mutation Observers instead. Example: https://github.com/sindresorhus/refined-github/blob/b695fc43a86965f31f55f740974608d4073dba1c/extension/content.js#L352-L353

**Comment by jgierer12** [comment]
Thanks, I had no idea about Mutation Observers 😅

**Comment by jgierer12** [comment]
Alright, using `MutationObserver` now. As @paulmolluzzo said, this also fixes the Markdown issue 👍

**Comment by paulmolluzzo** [comment]
@karanrajpal14 do you want to take a look at this PR?

File: extension/content.js [hunk 1]
```diff
@@ -179,10 +179,12 @@ function addPatchDiffLinks() {
...
 function removeDiffSigns() {
-	$('.blob-code-deletion .blob-code-inner, .blob-code-addition .blob-code-inner').each((index, element) => {
-		const textNode = element.childNodes[0];
-		textNode.nodeValue = textNode.nodeValue.replace(/^[+-]/, ' ');
+	const lines = $('.blob-code .blob-code-inner:not(.refined-github-diff-signs)');
+	lines.each((index, element) => {
+		const $element = $(element);
+		$element.html($element.html().replace(/^[+-]/, ''));
 	});
+	lines.addClass('refined-github-diff-signs');
```
Comment (paulmolluzzo) [inline extension/content.js]: Any reason you chose to break this whole thing out and not leave it as an inline `each`?

Also, any reason you're not adding the class within the `each`?

**Comment by karanrajpal14** [comment]
@paulmolluzzo doesn't seem to have fixed the issue for me. I still see the symbols for the diff I've linked to in the issue. I'm currently on 1.15.1.

File: extension/content.js (hunk 1 above)
Comment (jgierer12) [inline extension/content.js]: I think that was a left-over of something I had experimented with before. Moved it back inside the loop.

**Comment by jgierer12** [comment]
@karanrajpal14 You have to install the extension manually ([instructions](http://superuser.com/a/247654/6877)) from [here](https://github.com/jgierer12/refined-github/archive/bug/diff-signs.zip).

**Comment by jgierer12** [comment]
Just noticed that my `MutationObserver` doesn't work on review diffs in the timeline. (It throws an error on this page)

**Comment by karanrajpal14** [comment]
@jgierer12 I just installed the build that you linked to and it didn't help. Is it working fine on your end?

**Comment by jgierer12** [comment]
Can you check again if it works now?

**Comment by karanrajpal14** [comment]
Just checked. It's working fine now! Thank you so much! 👍

File: extension/content.js [hunk 2]
```diff
@@ -179,9 +179,10 @@ function addPatchDiffLinks() {
 }
 
 function removeDiffSigns() {
-	$('.blob-code-deletion .blob-code-inner, .blob-code-addition .blob-code-inner').each((index, element) => {
-		const textNode = element.childNodes[0];
-		textNode.nodeValue = textNode.nodeValue.replace(/^[+-]/, ' ');
+	$('.blob-code .blob-code-inner:not(.refined-github-diff-signs)').each((index, element) => {
```
Comment (sindresorhus) [inline extension/content.js]: `.blob-code` is too broad, it will match other elements too.

**Comment by sindresorhus** [comment]
Great stuff @jgierer12! Would you be interested in joining the project?

**Comment by jgierer12** [comment]
@sindresorhus Thanks, it would be great to join the project 😊

**Comment by sindresorhus** [comment]
@jgierer12 Woot! 🌈

File: extension/content.js [hunk 3]
```diff
@@ -179,10 +179,13 @@ function addPatchDiffLinks() {
...
-		const textNode = element.childNodes[0];
-		textNode.nodeValue = textNode.nodeValue.replace(/^[+-]/, ' ');
-	});
+	$('.blob-code-addition, .blob-code-deletion')
+		.find('.blob-code-inner:not(.refined-github-diff-signs)')
+		.each((index, element) => {
+			const $element = $(element);
+			$element.html($element.html().slice(1));
+			$element.addClass('refined-github-diff-signs');
+		});
```
Comment (sindresorhus) [inline extension/content.js]: I noticed now that this change makes some diffs unaligned.

For example: https://github.com/sindresorhus/globby/pull/40/files#diff-354f30a63fb0907d4ad57269548329e3

screen shot 2017-05-02 at 11 41 38

Comment (paulmolluzzo) [inline extension/content.js]: This shouldn't have been included in the PR is all. The MO was sufficient.

**Pull request merged**